WordPressKit/WordPressComRestApi.swift

```diff
@@ -285,18 +288,47 @@ open class WordPressComRestApi: NSObject {
                      success: @escaping SuccessResponseBlock,
                      failure: @escaping FailureReponseBlock) -> Progress? {
 
-        return request(method: .get, urlString: URLString, parameters: parameters, encoding: URLEncoding.default, success: success, failure: failure)
+        guard WordPressComRestApi.useURLSession else {
+            return request(method: .get, urlString: URLString, parameters: parameters, encoding: URLEncoding.default, success: success, failure: failure)
+        }
+
+        let progress = Progress.discreteProgress(totalUnitCount: 100)
+
+        Task { @MainActor in
+            let result = await self.perform(.get, URLString: URLString, parameters: parameters, fulfilling: progress)
+
+            switch result {
+            case let .success(response):
+                success(response.body, response.response)
+            case let .failure(error):
+                failure(error.asNSError(), error.response)
+            }
+        }
+
+        return progress
     }
 
     open func GETData(_ URLString: String,
                                          parameters: [String: AnyObject]?,
                                          completion: @escaping (Swift.Result<(Data, HTTPURLResponse?), Error>) -> Void) {
+        guard WordPressComRestApi.useURLSession else {
+            dataRequest(method: .get,
+                        urlString: URLString,
+                        parameters: parameters,
+                        encoding: URLEncoding.default,
+                        completion: completion)
+            return
+        }
+
+        Task { @MainActor in
+            let result: APIResult<Data> = await perform(.get, URLString: URLString, parameters: parameters)
 
-        dataRequest(method: .get,
-                    urlString: URLString,
-                    parameters: parameters,
-                    encoding: URLEncoding.default,
-                    completion: completion)
+            completion(
+                result
+                    .map { ($0.body, $0.response) }
+                    .mapError { error -> Error in error }
```

Comment:
It wasn't obvious to me why this `mapError` was necessary.

What to you think of adding a line such as

```suggestion
                    // The completion expects a result with any Error.
                    // We need to "downcast" the WordPressComRestApiEndpointError error in the APIResult.
                    .mapError { error -> Error in error }
```

Also, using `$0` would work, too. But, having the explicit function signature with `Error` might be clearer for the reader in this unusual scenario.